### PR TITLE
Remove unnecessary Frame set in Shape

### LINKED ISSUE #11704: Using Shapes causes continuous layout recalculation on Android
### Description

When a Shape is used in an Android app, the layout is constantly recalculated, significantly hurting performance.

### Steps to Reproduce

1. Create a Maui app
2. Add a Shape, for example:
```xml
<Line
    IsVisible="true"
    Stroke="Blue"
    X1="0"
    X2="100"
    Y1="0"
    Y2="50" />
```

3. Wrap it in a custom layout, for example:
```csharp
public class MyLayout : Layout
{
    protected override ILayoutManager CreateLayoutManager() => new MyLayoutManager(this);
}

public class MyLayoutManager : LayoutManager
{
    public MyLayoutManager(Microsoft.Maui.ILayout layout) : base(layout)
    {
    }

    public override Size ArrangeChildren(Rect bounds)
    {
        foreach (var view in Layout)
            view.Arrange(bounds);

        return new();
    }

    public override Size Measure(double widthConstraint, double heightConstraint)
    {
        return new(widthConstraint, heightConstraint);
    }
}
```

4. Run the app in Debug then put a breakpoint in `ArrangeChildren`. Notice how it is constantly being hit, even though the app is not doing anything.

### Link to public reproduction project repository

https://github.com/albi005/MauiApp39

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

all

### Did you find any workaround?

Not using Shapes, replacing them with GraphicsView instead.

### Relevant log output

_No response_

### LINKED ISSUE #12268: Infinite rendering loop when Rectangle placed in an AbsoluteLayout
### Description

The CPU becomes saturated when rectangles are placed in an AbsoluteLayout, when that is rendered as the *second* page in an AppShell. Don't ask me why the initial page doesn't show the effect.

### Steps to Reproduce

1. On a secondary page in an AppShell, add a Rectangle inside an AbsoluteLayout, and add LayoutBounds to it. May be observed using Android's HWUI rendering.

### Link to public reproduction project repository

[AbsoluteLayoutRectangleCPU.zip](https://github.com/dotnet/maui/files/10287313/AbsoluteLayoutRectangleCPU.zip)

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 13

### Did you find any workaround?

In one case, removing HorizontalOptions and VerticalOptions solved it, but not for all.

### Relevant log output

```shell
The only hint from the log is unending GC log lines:

[eterSidekick.XF] Explicit concurrent copying GC freed 35828(1616KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 40us,33us total 19.902ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35830(1615KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 49us,36us total 22.629ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35829(1616KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 43us,31us total 20.746ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35829(1615KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 61us,40us total 25.200ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35816(1616KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6175KB/12MB, paused 45us,39us total 23.028ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35824(1615KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 34us,31us total 17.068ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35836(1615KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 29us,36us total 18.352ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35842(1616KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6176KB/12MB, paused 46us,34us total 21.438ms
[eterSidekick.XF] Explicit concurrent copying GC freed 35836(1616KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 6175KB/12MB, paused 40us,34us total 20.120ms
```
```

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change

Shape has an unnecessary (and incorrect) extra setting of Frame which (under the right circumstances) causes infinite layout loops when interacting with the VisualDiagnosticOverlay on Android.

This set is left over from an earlier iteration of Border (which is a Shape) - it may have once served a purpose, or it may have always been wrong. The value it uses is incorrect (it blindly uses the view bounds, rather than the actual bounds of the Shape), and the Arrange method eventually corrects it, so the value is never really used to draw the shape. 

However, setting it triggers the Frame part mappings (X, Y, Width, Height) if the incorrectly computed value happens to be different from the correctly computed one. And triggering those mappings causes a call to RequestLayout. Which is fine unless you happen to have the VisualDiagnosticOverlay running, because that RequestLayout call bubbles up to the DecorView, and the layout event fired by DecorView causes the overlay to invalidate. Which triggers a re-draw of the Shape, and the process loops forever. 

tl;dr; - the set isn't required, and its side effects are awful.

### Issues Fixed

Fixes #11704
Fixes #12268

### CODE REVIEW

**Review by PureWeen — APPROVED** [review]

**Comment by BioTurboNick** [comment]
Thanks for looking into this and getting it done!

**Comment by rmarinho** [comment]
This broke the Border shapes 

image

**Comment by hartez** [comment]
> This broke the Border shapes

See the comment right before yours.
